docs/yarn_setup.md

```diff
@@ -173,6 +171,24 @@ You can check the number of TaskManagers in the JobManager web interface. The ad
 If the TaskManagers do not show up after a minute, you should investigate the issue using the log files.
 
 
+## Run a single Flink job on Hadoop YARN
+
+The documentation above describes how to start a Flink cluster within a Hadoop YARN environment.
+It is also possible to launch Flink within YARN only for executing a single job.
+
+To deploy a job to a per-job YARN cluster, set the master name to `yarn-cluster`.
+Please note that the client then expects the `-n` value to be set (number of TaskManagers).
```

Comment:
Thank you for having a look at the PR.
Yes, you are right. I will update the pull request and add a fix for FLINK-1385 to it as well.